File: backend/src/main/java/com/gestionstock/backend/service/fournisseur/VenteService.java
package com.gestionstock.backend.service.fournisseur;

import com.gestionstock.backend.dto.fournisseur.LigneVenteDTO;
import com.gestionstock.backend.dto.fournisseur.VenteDTO;
import com.gestionstock.backend.entity.enums.ModePaiement;

import java.util.List;

/**
 * Interface du service Vente.
 * Définit les opérations métier pour la gestion des ventes (point de vente).
 */
public interface VenteService {

    /** Récupère toutes les ventes triées par date décroissante */
    List<VenteDTO> getAllVentes();

    /** Récupère une vente par son identifiant (avec ses lignes) */
    VenteDTO getVenteById(Long id);

    /** Récupère les ventes d'un client donné */
    List<VenteDTO> getVentesByClient(Long clientId);

    /** Crée une nouvelle vente (point de vente) avec ses lignes, met à jour le stock */
    VenteDTO createVente(Long clientId, ModePaiement modePaiement, List<LigneVenteDTO> lignes);

    /** Supprime une vente par son identifiant */
    void deleteVente(Long id);
}

File: frontend/src/app/core/services/vente.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';
import { Vente, VenteRequest } from '../models/vente.model';
import { API_BASE_URL } from './api.config';

const API_URL = API_BASE_URL;

/**
 * Service Angular pour la gestion des ventes (point de vente).
 * Communique avec le backend via les endpoints REST /api/ventes.
 */
@Injectable({
  providedIn: 'root'
})
export class VenteService {

  private http = inject(HttpClient);

  /** Récupère toutes les ventes */
  getAllVentes(): Observable<Vente[]> {
    return this.http.get<Vente[]>(`${API_URL}/ventes`);
  }

  /** Récupère une vente par son ID (avec ses lignes) */
  getVenteById(id: number): Observable<Vente> {
    return this.http.get<Vente>(`${API_URL}/ventes/${id}`);
  }

  /** Récupère les ventes d'un client */
  getVentesByClient(clientId: number): Observable<Vente[]> {
    return this.http.get<Vente[]>(`${API_URL}/ventes/client/${clientId}`);
  }

  /** Crée une nouvelle vente */
  createVente(request: VenteRequest): Observable<Vente> {
    return this.http.post<Vente>(`${API_URL}/ventes`, request);
  }

  /** Supprime une vente */
  deleteVente(id: number): Observable<void> {
    return this.http.delete<void>(`${API_URL}/ventes/${id}`);
  }
}
